### ISSUE TITLE
offline cache fails with relative --cwd

### ISSUE THREAD

**Issue by mark-buer** [yarnpkg/yarn#5327]
**Do you want to request a *feature* or report a *bug*?**
Bug.

**What is the current behavior?**
`yarn install --cwd some/relative/path` fails to fill the offline cache directory

whereas...

`yarn install --cwd /some/absolute/path` successfully fills the offline cache directory.

**If the current behavior is a bug, please provide the steps to reproduce.**

```sh
#!/bin/sh

# config
PARENT_PATH="${PWD}/parent"
SUB_PATH="${PARENT_PATH}/sub"
OFFLINE_MIRROR_PATH="${PARENT_PATH}/.yarn-offline-mirror"

# setup the test scenario
mkdir -p "$SUB_PATH"
cd "$SUB_PATH"
yarn init --yes
yarn add semver
rm -rf node_modules

# enable offline mirror
echo yarn-offline-mirror "${OFFLINE_MIRROR_PATH}" >"${PARENT_PATH}/.yarnrc"

# install from within sub directory
yarn install
echo ">> after install from sub directory, offline mirror contains..."
ls "$OFFLINE_MIRROR_PATH"
# as expected, offline mirror has package within it

# clean up
rm -rf "${SUB_PATH}/node_modules"
rm -rf "$OFFLINE_MIRROR_PATH"
echo ">> after cleanup, offline mirror contains..."
ls "$OFFLINE_MIRROR_PATH"

cd "$PARENT_PATH"

yarn install --cwd "$SUB_PATH"
echo ">> after install from parent directory (absolute path), offline mirror contains..."
ls $OFFLINE_MIRROR_PATH
# as expected, offline mirror has package within it

# clean up
rm -rf "${SUB_PATH}/node_modules"
rm -rf "$OFFLINE_MIRROR_PATH"
echo ">> after cleanup, offline mirror contains..."
ls "$OFFLINE_MIRROR_PATH"

yarn install --cwd sub
echo ">> after install from parent directory (relative path), offline mirror contains..."
ls $OFFLINE_MIRROR_PATH
# error, offline mirror has no package within it
```

**What is the expected behavior?**

Relative `--cwd` paths should work identically to absolute `--cwd` paths.

**Please mention your node.js, yarn and operating system version.**

node --version => v8.9.1
yarn --version => 1.3.2
macOS 10.13.3

**Comment by mark-buer** [comment]
Related to #4984?
Thus might be resolved with #4991?

**Comment by mark-buer** [comment]
I applied the changes in #4991 to my local yarn, and can confirm it does fix the issue.
